aws/validators.go

```diff
@@ -1125,8 +1126,16 @@ func validateAppautoscalingCustomizedMetricSpecificationStatistic(v interface{},
 
 func validateAppautoscalingPredefinedMetricSpecification(v interface{}, k string) (ws []string, errors []error) {
 	validMetrics := []string{
-		"DynamoDBReadCapacityUtilization",
-		"DynamoDBWriteCapacityUtilization",
+		applicationautoscaling.MetricTypeAlbrequestCountPerTarget,
+		applicationautoscaling.MetricTypeDynamoDbreadCapacityUtilization,
+		applicationautoscaling.MetricTypeDynamoDbwriteCapacityUtilization,
+		applicationautoscaling.MetricTypeEc2spotFleetRequestAverageCpuutilization,
+		applicationautoscaling.MetricTypeEc2spotFleetRequestAverageNetworkIn,
+		applicationautoscaling.MetricTypeEc2spotFleetRequestAverageNetworkOut,
+		applicationautoscaling.MetricTypeEcsserviceAverageCpuutilization,
+		applicationautoscaling.MetricTypeEcsserviceAverageMemoryUtilization,
+		applicationautoscaling.MetricTypeRdsreaderAverageCpuutilization,
+		applicationautoscaling.MetricTypeRdsreaderAverageDatabaseConnections,
```

Comment:
Something tells me this list is going to continue to grow and we just shouldn't be validating it at all, instead of playing catch up with Amazon... 🤔

Reply:
I agree with you for the most part. Its a challenge of better user experience (plan-time validation and requiring provider updates) vs development time (SDK not providing a quick way to get the full list of them). Since this PR is approved and we're planning on releasing today, I'll bring this one in as-is to at least fix the current state of the world, then will followup with a PR to remove the validation completely. We'll see if it lands in time for today as well.

As a separate pet project I will likely try to automatically `go generate` these from the SDK as I personally have found the plan-time validation to helpful and this type of generation would be applicable in a few other places. Then again I also always kept my AWS provider up to date in my environments.

Reply:
As noted, followup PR to remove validation (if we want to): #3141